src/flb_io.c

```diff
@@ -174,6 +174,23 @@ int flb_io_net_connect(struct flb_connection *connection,
     return 0;
 }
 
+static void net_io_propagate_critical_error(
+                struct flb_connection *connection)
+{
+    switch (errno) {
+    case EBADF:
+    case ECONNRESET:
+    case EDESTADDRREQ:
+    case ENOTCONN:
+    case EPIPE:
+    case EACCES:
+    case EIO:
```

Comment:
Looks good, but wondering how you got a list of critical/unrecoverable errors?

EIO seems pretty broad and may not be unrecoverable, though I don't know anything about these errors beside what is stated in the manual - https://man7.org/linux/man-pages/man3/errno.3.html:
```
Input/output error
```

Reply:
I got those from the man page entries for send and recv but you are probably right about EIO.
Just keep in mind that the only places where this mechanism is used are the socket io functions in `flb_io.c`.

Reply:
As suggested I removed `EIO` and added `ENOTTY`.